stores/message.js

```diff
@@ -173,6 +174,16 @@ export const useMessageStore = defineStore({
     },
     async fetchByUser(userid, active, force) {
       let messages = []
+      const findByIdAndUpdate = (messages) => {
+        messages.forEach((message) => {
+          const i = this.byUserList[userid].findIndex(curMessage => curMessage.id === message.id)
+          if (i !== -1) {
+            this.byUserList[userid][i] = message
+          } else {
+            this.byUserList[userid].push(message)
+          }
+        })
+      }
```

Comment:
I don't think this has quite the same effect as the old code.  It's possible that a message has been deleted on the server side (by a moderator).  I think this new code would leave that deleted message in store.

I always have to think hard about replacing the whole array, and whether it works reactively.  But could we use `Vue.set(this.byUserList, userid, messages)`?  That would be neater than a loop.

Reply:
There is no more Vue.set in vue 3 :)

So, my idea was to update those messages we have already in store when we receive new messages from the API. And, if we have 50 messages loaded and then the response comes containing 10 items, we would update data for those 10 items and leave the other 40 intact. I was thinking that the response might be paginated, if not and we always get all user messages, we can surely just replace what we have in the store.

Reply:
I've pushed the fix which updates per item from response only for `active: true`. As I understand, when `active: false` we always get all the messages user have

Reply:
The `active` parameter controls whether the API returns messages which have not yet been TAKEN/RECEIVED (active = true) or all messages (active = false).  It's not relevant here, so you should revert that change.

The response isn't paginated. So yes, you can replace all the ones you currently have, but you also need to remove the ones that have no longer been returned from the server.  This would be true even if the response was paginated.

Suppose you initially have messages A, B, C in store on the client.  The message B is deleted on the server while the client still has the page open.  The user fetches messages again, and will receive messages A and C.  The code will loop through and update messages A and C, but message B will remain in the store.  That's the bug.

If this still isn't clear, then let's have a Slack call.

Reply:
@misterpekert I think `findByIdAndUpdate` still has this issue.  If a message is no longer returned from the server, it will be left in the store.

Reply:
Yeah, you’re right. I think we need to always fetch all posts in this case with `active: false`, otherwsie we might end up with missmatch between client and server. I’ll update the PR